### PR TITLE
Adds Open vSwitch bond management module

### PR DESCRIPTION
**Pull request by busterswt**
##### SUMMARY
Adds a new Open vSwitch bond management module with support for multiple other_config and set commands necessary for multiple interfaces, including DPDK-enabled interfaces.

Based on the existing openvswitch_port module.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
openvswitch_bond

##### ADDITIONAL INFORMATION
The existing Open vSwitch modules do not address the configuration and management of bonded interfaces. This module allows users to create and add bonded interfaces to existing bridges by leveraging the ```ovs-vsctl add-bond``` command and associated configuration options.

### CODE REVIEW

File: plugins/modules/openvswitch_bond.py [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+      - "0000:04:00.0"
+      - "0000:04:00.1"
+    lacp: active
+    set:
+      - "interface 0000:04:00.0 type=dpdk options:dpdk-devargs=0000:04:00.0"
+      - "interface 0000:04:00.1 type=dpdk options:dpdk-devargs=0000:04:00.1"
+    state: present
+"""
+
+RETURN = """ # """
```
Comment (Qalthos) [inline plugins/modules/openvswitch_bond.py:139]: Please add details about the returned `commands` list

**Review by Qalthos — CHANGES_REQUESTED** [review]
You will also need at minimum to add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) and some unit tests for the module before this can be merged.

File: plugins/modules/openvswitch_bond.py (hunk 1 above)
Comment (busterswt) [reply plugins/modules/openvswitch_bond.py]: Thanks. Should be addressed now.

File: tests/unit/modules/network/ovs/test_openvswitch_bond.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            side_effect_with_fixture_loaded = (rc, out, err)
+            test_side_effects.append(side_effect_with_fixture_loaded)
+        self.run_command.side_effect = test_side_effects
+
+        self.get_bin_path.return_value = "/usr/bin/ovs-vsctl"
+
+    def test_openvswitch_bond_absent_idempotent(self):
+        set_module_args(dict(state="absent", bridge="bond-br", port="bond0"))
+        self.execute_module(
+            test_name="test_openvswitch_bond_absent_idempotent"
```
Comment (Qalthos) [inline tests/unit/modules/network/ovs/test_openvswitch_bond.py:105]: I'm assuming this is supposed to result in no commands issued, you can ensure that with `commands=[]` (vs not specifying `commands`, which makes no assumptions about what commands might be returned)
```suggestion
            commands=[],
            test_name="test_openvswitch_bond_absent_idempotent"
```

File: tests/unit/modules/network/ovs/test_openvswitch_bond.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            dict(
+                state="present",
+                bridge="bond-br",
+                port="bond0",
+                external_ids={"foo": "bar"},
+                other_config={"bond-detect-mode": "miimon"},
+            )
+        )
+        self.execute_module(
+            test_name="test_openvswitch_bond_present_idempotent"
```
Comment (Qalthos) [inline tests/unit/modules/network/ovs/test_openvswitch_bond.py:128]: Same here
```suggestion
            commands=[],
            test_name="test_openvswitch_bond_present_idempotent"
```

**Review by Qalthos — CHANGES_REQUESTED** [review]
Looks fine otherwise

File: tests/unit/modules/network/ovs/test_openvswitch_bond.py (hunk 2 above)
Comment (busterswt) [reply tests/unit/modules/network/ovs/test_openvswitch_bond.py:106]: That's correct, no commands are expected in these cases. Thanks for the assist!

**Review by Qalthos — APPROVED** [review]

**Review by NilashishC — APPROVED** [review]

**Pull request merged**